### DISCUSSION TITLE
How to split stack of patches in 2 (or more) branches?

### DISCUSSION THREAD

**Discussion by tino** [stacked-git/stgit#498] [Q&A] [answered]
Is there an easy way to move patches from one branch (or "stacks") to another? Sometimes I want to split the feature I'm working on, and for now I've done that with creating a new branch at the top of the stack, uncommitting all on that stack (so I have two of the same stacks), and then modifying both. It feels a bit unwieldy though. 

How do others do this? Would a `stg send-to my-other-branch` make sense?

**Comment by jpgrayson** [comment, marked as answer]
The following script is helpful for me. Here's the workflow:

* Develop patches on main stack.
* Eventually, a subset of those patches will be ready to become a PR/MR. To appease GitHub/GitLab's workflow, I need those particular patches to be on their own branch.
* In main stack, apply the patches to be carved onto their own branch.
* Run this script, providing the new branch name as the argument.

```bash
#!/usr/bin/env bash

set -euo pipefail

top=$(stg top)
newbranch=${1:-${top/-/'/'}}
curbranch=$(stg branch)

set -x

stg branch -C "$newbranch"
stg delete -U
stg branch "$curbranch"
stg delete "..$top"
stg rebase "$newbranch"
stg branch "$newbranch"
```

After the script runs, a new branch exists with only the selected (applied) patches and `main` is rebased onto the new branch with the selected patches removed.

For PR/MR workflows, the newly created branch is ready to be pushed.

And development of work-in-progress patches on `main` can be resumed.

-----

W.r.t. the workflow you describe, not sure if you know about `stg branch -C/--clone`. That's kind of the foundational piece to my script. Might be preferable to using `stg uncommit`.

**Reply by tino** [reply]
Ah, I indeed wasn't aware of `--clone` for `stg branch`. That should help already!
